packages/fetchai/connections/http_client/connection.py

```diff
@@ -0,0 +1,276 @@
+        self.in_queue = None  # type: Optional[asyncio.Queue]  # pragma: no cover
+        self.loop = None  # type: Optional[asyncio.AbstractEventLoop]  # pragma: no cover
+        self.excluded_protocols = excluded_protocols
+        self.is_stopped = True
+        logger.info("Initialised the HTTP client channel")
+
+    def connect(self):
+        """Connect."""
+        if self.is_stopped:
+            self.is_stopped = False
+            try:
+                logger.info(self.address)
+                response = requests.request(
+                    method="GET", url=f"http://{self.address}:{self.port}/status"
+                )
+                logger.debug("Status code is: {}".format(response.status_code))
+                assert response.status_code == 200, "Connection failed."
+            except Exception as e:  # pragma: no cover
+                logger.warning(str(e))
+                raise
+
+    def send(self, envelope: Envelope) -> None:
+        """
+        Send the envelope as a HTTP request, wait for its response, translate the response into an envelop, and sent it back to the agent.
+
+        :param envelope: the envelope
+        :return: None
+        """
+        if self.excluded_protocols is not None:
+            if envelope.protocol_id in self.excluded_protocols:
+                logger.error(
+                    "This envelope cannot be sent with the oef connection: protocol_id={}".format(
+                        envelope.protocol_id
+                    )
+                )
+                raise ValueError("Cannot send message.")
+
+            if envelope is not None:
+                http_message = cast(
+                    HttpMessage, HttpSerializer().decode(envelope.message)
+                )
+                if http_message.performative == HttpMessage.Performative.REQUEST:
+                    response = requests.request(
+                        method=http_message.method,
+                        url=http_message.url,
+                        headers=http_message.headers,
+                        data=http_message.bodyy,
+                    )
+                    envelope = self.to_envelope(response)
+                    self.loop.call_soon_threadsafe(self.in_queue.put_nowait, envelope)
+                else:
+                    raise ValueError("The HTTPMessage performative must be a REQUEST.")
+            else:
+                raise ValueError("The Envelop object to be sent is None.")
+
+    def to_envelope(self, http_response: requests.models.Response) -> Envelope:
+        """
+        Process HTTP response by packaging it into an Envelope and sending it to in-queue.
+
+        :param http_response: the http response object
+        """
+
+        # uri = URI(self.full_url_pattern)
+        # context = EnvelopeContext(connection_id=connection_id, uri=uri)
+        http_message = HttpMessage(
+            dialogue_reference=("", ""),
```

Comment:
I think the dialogue reference, target and message id should concur with those of the request envelope.